### DISCUSSION TITLE
Why do we need a daemonset for restic plugin. Why not a deployment ??

### DISCUSSION THREAD

**Discussion by vishumindtree** [vmware-tanzu/velero#4132] [Community support Q&A] [answered]
I am using restic with velero to take backups of PVCs but by default it has a daemonset object of K8s cluster.

So if its not required to run on all Kubernetes nodes ,then there is no need to have daemonset for this .We can simply have a deployment K8s object for this.

**Comment by vishumindtree** [comment]
Team ,

Please provide your inputs on this as soon as possible . Thanks !

**Comment by sseago** [comment, marked as answer]
It runs as a DaemonSet because the restic pod uses mount propagation to access a pod's volumes for backup/restore. This means that when a pod is being backed up or restored, for the restic pod to access the volume in question, it must be running on the same node as the pod being backed up or restored. Creating a DaemonSet ensures that every node that might have pods running in it will have an available restic pod which is able to access the volume.

**Reply by vishumindtree** [reply]
Sure ! thanks ! it makes sense .
